webapp/components/_new/generic/ResponsiveImage/ResponsiveImage.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <img :data-src="lazySrc" :data-srcset="lazySrcset" :style="style" class="responsive-image lozad" />
+</template>
+
+<script>
+import lozad from 'lozad'
+
+export default {
+  name: 'ResponsiveImage',
+  props: {
+    backgroundColor: {
+      type: String,
+      default: '#efefef',
+    },
+    height: {
+      type: Number,
+      default: null,
+    },
+    lazySrc: {
+      type: String,
+      default: null,
+    },
+    lazySrcset: {
+      type: String,
+      default: null,
+    },
+    width: {
+      type: Number,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      loading: true,
+    }
+  },
+  computed: {
+    aspectRatio() {
+      if (!this.width || !this.height) return null
+      return (this.height / this.width) * 100
+    },
+    style() {
+      const style = { backgroundColor: this.backgroundColor }
+      if (this.width) style.width = `${this.width}px`
+      const applyAspectRatio = this.loading && this.aspectRatio
+      if (applyAspectRatio) {
+        style.height = 0
+        style.paddingTop = `${this.aspectRatio}%`
+      }
+
+      return style
+    },
+  },
+  mounted() {
+    // As soon as the <img> element triggers
+    // the `load` event, the loading state is
+    // set to `false`, which removes the apsect
+    // ratio we've applied earlier.
+    const setLoadingState = () => {
+      this.loading = false
+    }
+    this.$el.addEventListener('load', setLoadingState)
+    // We remove the event listener as soon as
+    // the component is destroyed to prevent
+    // potential memory leaks.
+    this.$once('hook:destroyed', () => {
+      this.$el.removeEventListener('load', setLoadingState)
+    })
+
+    // We initialize Lozad.js on the root
+    // element of our component.
+    const observer = lozad(this.$el)
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-01-28T23:06:42Z" title="Wednesday, January 29th 2020, 12:06:42 am +01:00">Jan 29, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,87 @@
+<template>
+  <img :data-src="lazySrc" :data-srcset="lazySrcset" :style="style" class="responsive-image lozad" />
+</template>
+
+<script>
+import lozad from 'lozad'
+
+export default {
+  name: 'ResponsiveImage',
+  props: {
+    backgroundColor: {
+      type: String,
+      default: '#efefef',
+    },
+    height: {
+      type: Number,
+      default: null,
+    },
+    lazySrc: {
+      type: String,
+      default: null,
+    },
+    lazySrcset: {
+      type: String,
+      default: null,
+    },
+    width: {
+      type: Number,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      loading: true,
+    }
+  },
+  computed: {
+    aspectRatio() {
+      if (!this.width || !this.height) return null
+      return (this.height / this.width) * 100
+    },
+    style() {
+      const style = { backgroundColor: this.backgroundColor }
+      if (this.width) style.width = `${this.width}px`
+      const applyAspectRatio = this.loading && this.aspectRatio
+      if (applyAspectRatio) {
+        style.height = 0
+        style.paddingTop = `${this.aspectRatio}%`
+      }
+
+      return style
+    },
+  },
+  mounted() {
+    // As soon as the <img> element triggers
+    // the `load` event, the loading state is
+    // set to `false`, which removes the apsect
+    // ratio we've applied earlier.
+    const setLoadingState = () => {
+      this.loading = false
+    }
+    this.$el.addEventListener('load', setLoadingState)
+    // We remove the event listener as soon as
+    // the component is destroyed to prevent
+    // potential memory leaks.
+    this.$once('hook:destroyed', () => {
+      this.$el.removeEventListener('load', setLoadingState)
+    })
+
+    // We initialize Lozad.js on the root
+    // element of our component.
+    const observer = lozad(this.$el)
```


:thinking: what does that do?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-29T09:05:07Z" title="Wednesday, January 29th 2020, 10:05:07 am +01:00">Jan 29, 2020</time>_

---


it's how to set up `lozad`, have you had a chance to look at the documentation of how it works?